src/skydb.ts

```diff
@@ -346,23 +361,13 @@ export async function getOrCreateRegistryEntry(
   }
   const file = new File([JSON.stringify(fullData)], `dk:${dataKeyHex}`, { type: "application/json" });
 
-  // Start file upload, do not block.
+  // Do file upload.
   const uploadOpts = extractOptions(opts, defaultUploadOptions);
-  const skyfilePromise: Promise<UploadRequestResponse> = client.uploadFile(file, uploadOpts);
-
-  // Fetch the current value to find out the revision.
-  //
-  // Start getEntry, do not block.
-  const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
-  const entryPromise: Promise<SignedRegistryEntry> = client.registry.getEntry(publicKey, dataKey, getEntryOpts);
+  const skyfile = await client.uploadFile(file, uploadOpts);
 
-  // Block until both getEntry and uploadFile are finished.
-  const [signedEntry, skyfile] = await Promise.all<SignedRegistryEntry, UploadRequestResponse>([
-    entryPromise,
-    skyfilePromise,
-  ]);
-
-  const revision = getRevisionFromEntry(signedEntry.entry);
+  // Get the new revision by incrementing the one in the cache, or use 0 if not cached.
+  let revision: bigint = client.revisionNumberCache[getCacheKey(publicKey, dataKey)] ?? BigInt(-1);
```

Comment:
I might be misunderstanding because it goes completely against this change but why are we assuming that we have the registry revision cached if it exists and if not then the registry entry is empty? Previous implementation had explicit `client.registry.getEntry` call to look for the entry just before writing to registry to either pull the version or find out that it doesn't exist. `revisionNumberCache` will only have the value if `getJSON` is called and even if, we're not sure that the cached data is not stale.

I'm going to mark this PR as changes requested even though it's a question but I want to make sure it's blocked until resolved.